tests/integration_tests/test_grpc_only.py

```diff
@@ -0,0 +1,128 @@
+import base64
+import json
+import subprocess
+import time
+from pathlib import Path
+
+import pytest
+import requests
+from pystarport import ports
+
+from .network import setup_custom_ethermint
+from .utils import (
+    CONTRACTS,
+    decode_bech32,
+    deploy_contract,
+    supervisorctl,
+    wait_for_block,
+    wait_for_port,
+)
+
+
+@pytest.fixture(scope="module")
+def custom_ethermint(tmp_path_factory):
+    path = tmp_path_factory.mktemp("grpc-only")
+
+    # reuse rollback-test config because it has an extra fullnode
+    yield from setup_custom_ethermint(
+        path,
+        26400,
+        Path(__file__).parent / "configs/rollback-test.jsonnet",
+    )
+
+
+def grpc_eth_call(port: int, args: dict, chain_id=None, proposer_address=None):
+    """
+    do a eth_call through grpc gateway directly
+    """
+    params = {
+        "args": base64.b64encode(json.dumps(args).encode()).decode(),
+    }
+    if chain_id is not None:
+        params["chain_id"] = str(chain_id)
+    if proposer_address is not None:
+        params["proposer_address"] = str(proposer_address)
+    return requests.get(
+        f"http://localhost:{port}/ethermint/evm/v1/eth_call", params
+    ).json()
+
+
+def test_grpc_mode(custom_ethermint):
+    """
+    - restart a fullnode in grpc-only mode
+    - test the grpc queries all works
+    """
+    w3 = custom_ethermint.w3
+    contract = deploy_contract(w3, CONTRACTS["TestChainID"])
+    assert 9000 == contract.caller.currentChainID()
+
+    msg = {
+        "to": contract.address,
+        "data": contract.encodeABI(fn_name="currentChainID"),
+    }
+    api_port = ports.api_port(custom_ethermint.base_port(1))
+    # in normal mode, grpc query works even if we don't pass chain_id explicitly
+    success = False
+    max_retry = 3
+    sleep = 1
+    for i in range(max_retry):
+        rsp = grpc_eth_call(api_port, msg)
+        ret = rsp["ret"]
+        valid = ret is not None
+        if valid and 9000 == int.from_bytes(base64.b64decode(ret.encode()), "big"):
```

Comment:
[test](https://github.com/crypto-org-chain/ethermint/actions/runs/4362229790/jobs/7626882060) was not stable, is it ok to merge the test [fix](https://github.com/crypto-org-chain/ethermint/pull/216) together? @yihuang

Reply:
sure 👍